### PR TITLE
Add a L0 and L1 fits for other TM (not BSD or QL or HK) using a DefaultProduct

### PR DESCRIPTION
**Pull request by samaloney**
* Create fits files based on the parameter names
* For L1 put in human readable subfolders, e.g. events, diagnostics, tm-verify ...

### CODE REVIEW

File: stixcore/products/product.py [hunk 1]
```diff
@@ -315,3 +315,199 @@ def __repr__(self):
...
+class GenericProduct(BaseProduct):
+    def __init__(self, *, service_type, service_subtype, ssid, control, data,
+                 idb_versions=defaultdict(SCETimeRange), **kwargs):
+        """
+        Generic product composed of control and data
+
+        Parameters
+        ----------
+        control : stix_parser.products.quicklook.Control
+            Table containing control information
```
Comment (nicHoch) [inline stixcore/products/product.py:329]: this is nice that some of the methods are now move to this super class.

what is your definition difference between Base and Generic Product?

File: stixcore/products/product.py [hunk 2]
```diff
@@ -315,3 +315,199 @@ def __repr__(self):
...
+        times = SCETime(control['scet_coarse'], control['scet_fine'])
+        scet_timerange = SCETimeRange(start=times[0], end=times[-1])
+
+        # Data
+        data = Data()
+        data['time'] = times
+        data['timedel'] = SCETimeDelta(0, 0)
+
+        reshape = True if {'NIX00103', 'NIX00104'}.intersection(
+            packets.data[0].__dict__.keys()) else False
```
Comment (nicHoch) [inline stixcore/products/product.py:486]: compact but not so easy to read

**Review by nicHoch — COMMENTED** [review]
looks good.
i wonder why you changed the datetimes in the idbVersionHistory.json

File: stixcore/data/idb/idbVersionHistory.json [hunk 3]
```diff
@@ -11,14 +11,14 @@
...
-                "validityPeriodOBT":[{"coarse":642159994, "fine":0}, {"coarse":694310300, "fine":0}]
+                "validityPeriodOBT":[{"coarse":642159994, "fine":0}, {"coarse":677766880, "fine":0}]
         },
         {
                 "version":"2.26.35",
                 "aswVersion":181,
                 "comment": "return this for the next future",
                 "validityPeriodUTC":["2022-01-01T00:00:00", "2030-01-01T00:00:00"],
-                "validityPeriodOBT":[{"coarse":694310300, "fine":0}, {"coarse":4294967295, "fine":0}]
+                "validityPeriodOBT":[{"coarse":677766880, "fine":0}, {"coarse":4294967295, "fine":0}]
```
[the remark this answers is not in the document]
Comment (samaloney) [reply stixcore/data/idb/idbVersionHistory.json:21]: No is the answer but 694310300 seem was to far in the future - I looked at the HK tm and for the change between v180 and v181 and used that?

File: stixcore/products/product.py (hunk 1 above)
Comment (samaloney) [reply stixcore/products/product.py:329]: Yea more work can be done here, the idea was the `Base` is inherited by other products but `GenericProduct` is not?

File: stixcore/io/fits/processors.py [hunk 4]
```diff
@@ -491,11 +490,21 @@ def write_fits(self, product):
...
-            energy_hdu = table_to_hdu(energy_enc)
-            energy_hdu.name = 'ENERGIES'
+            hdus = [primary_hdu, control_hdu, data_hdu, idb_hdu]
+
+            if getattr(prod, 'get_energies', False) is not False:
+                elow, ehigh = prod.get_energies()
+                energies = QTable()
+                energies['channel'] = range(len(elow))
+                energies['e_low'] = elow * u.keV
+                energies['e_high'] = ehigh * u.keV
```
[the remark this answers is not in the document]
Comment (samaloney) [reply stixcore/io/fits/processors.py:500]: Created new issue for this #135

**Review by nicHoch — COMMENTED** [review]
all good

**Review by nicHoch — APPROVED** [review]
all good

**Pull request merged**